Add USPTO documentation

AggelosMargkas opened the pull request:
Add my first example of results of data that run through the whole USPTO data.

Add the sql query into the uspto examples file. This SQL query is designed to extract information about the applicant population by country and year, focusing on the top 5 countries in terms of patent counts for the year 2022.

Add the graph representing the evolution of the population and show it on the README.

dspinellis reviewed (changes requested):
Nice! I like the clear SQL code. Please also add the Python script that generates the chart.

AggelosMargkas commented:
> Nice! I like the clear SQL code. Please also add the Python script that generates the chart.

Added that, should I add the csv data file as well?

dspinellis commented:
Great! The CSV data is a generated file. We add the chart (which is also generated) so that it can appear in the README file. I'll ensure that the example can run automatically as part of integration testing and get back to you, if needed.

dspinellis commented on docs/app-eg.rst:
```diff
@@ -61,3 +61,5 @@ The following examples are available.
    works <https://github.com/dspinellis/alexandria3k/tree/main/examples/tawfik>`__
 -  `Fully-indexed Crossref database
    sample <https://github.com/dspinellis/alexandria3k/tree/main/examples/sample>`__
+-  `Top 5 USPTO applicants population by country and year  
```
Top five US Patent Office applications by …

dspinellis commented on docs/cli-eg.rst:
```diff
@@ -63,6 +63,54 @@ where ``count-year-type.sql`` contains:
    SELECT type AS name, Sum(number) FROM counts
      GROUP BY type
 
+Count USPTO top 5 applicant countries from 2005 to 2022
```
Obtain top-five US Patent Office application countries

dspinellis commented on docs/cli-eg.rst:
```diff
@@ -98,6 +146,36 @@ you can re-seed the random number generator for each sample with
+the weekly issued patents (path) or for the patents(container) which
+are located inside the issued file. The second value of the tuple can
+be an USPTO Zip file path or a container respectively.
+
```
For the USPTO dataset sampling is performed through a 
provided tuple argument named **data**. The tuple's first value is
a designator string that will be either "path" or "container".
Thus, sampling can be done either on compressed files represented
through the weekly issued patents (path) or on the patents' full text
(container) located within the issued file.
The tuple's second value is a USPTO Zip file path or the patent's text, respectively.

dspinellis commented on docs/cli-eg.rst:
```diff
@@ -98,6 +146,36 @@ you can re-seed the random number generator for each sample with
+.. note::
+   How to access the 
+   sampling tuple? It is already defined as a variable inside *alexandria3k* and
+   is named **data**. So, the first value of the tuple will data[0].
```
I covered it above, please remove the note.

dspinellis commented on docs/downloading.rst:
```diff
@@ -41,6 +41,44 @@ download the *summary file* of the ORCID Public Data File
+~~~~~~~~~~
+
+You can populate a database with the bibliographic text (front page) of each
+patent application published weekly from 2003 to present. This data are published
```
These data

dspinellis commented on docs/downloading.rst:
```diff
@@ -41,6 +41,44 @@ download the *summary file* of the ORCID Public Data File
+
+You can populate a database with the bibliographic text (front page) of each
+patent application published weekly from 2003 to present. This data are published
+from the US Patent and Trademark Office (USPTO) and hence the data source acronym inside *alexandria3k*.
```
inside → used in

dspinellis commented on docs/downloading.rst:
```diff
@@ -41,6 +41,44 @@ download the *summary file* of the ORCID Public Data File
+   ├── 2010
+       ├── ipgb20100223_wk08.zip
+       └── ipgb20100302_wk09.zip
+
```
Add:
```
The shell script *examples/uspto/fetch.sh* can be used to download all USPTO data.
```

dspinellis commented on examples/uspto/README.md:
```diff
@@ -0,0 +1,5 @@
+# Applicants Population by Year and Country
+
+The graph illustrates the calculated population of applicants who are listed in the records of the US Patent and Trademark Office, focusing specifically on the top five countries of the year 2022.
```
The chart illustrates the number of applicants listed in the records of the US Patent and Trademark Office,
tracking those from the top five countries of the year 2022.

dspinellis commented on examples/uspto/README.md:
```diff
@@ -0,0 +1,5 @@
+
+The graph illustrates the calculated population of applicants who are listed in the records of the US Patent and Trademark Office, focusing specifically on the top five countries of the year 2022.
+
+![Applicants Population by Country and year for the Top 5 Countries of 2022](./top5applicantcountries.png)
```
![Number of Applicants by Country and Year for the Top-Five Countries of 2022](./top5applicantcountries.png)Top-Five

dspinellis commented on examples/uspto/plot_applicants_data.py:
```diff
@@ -0,0 +1,46 @@
+# Show the plot
+plt.grid(True)
+plt.tight_layout()
+plt.show()
```
Shouldn't the last line save the data into a file?
Can the file be SVG, rather than PNG?

dspinellis reviewed (changes requested):
So nice to see this contribution! I suggested a few changes.

AggelosMargkas commented:
> So nice to see this contribution! I suggested a few changes.

Hello👋 and apologies for my late reply. I updated the changes and I believe they are ready to be seen. I changed the image format accordingly. 

I have this error on the CI log for the installation of the rdbunit.

```
Run git clone --depth=1 https://github.com/dspinellis/rdbunit.git
Cloning into 'rdbunit'...
make: *** No rule to make target 'install'.  Stop.
Error: Process completed with exit code 2.
```

Any idea how to resolve this?
